src/app/components/room/room.component.ts

```diff
@@ -80,20 +105,50 @@ export class RoomComponent implements OnInit {
   }
 
   get canPost(): boolean {
-    return this.messageInput.trim().length != 0;
+    return this.messageInput.trim().length != 0 && this.titleInput.trim().length != 0;
   }
 
 
   timePosted(postedDate: Date):string {
-    let date = new Date(postedDate);
     let currentDate = new Date();
+    postedDate = new Date(postedDate);
 
-
-    let days = Math.floor((currentDate.getTime() - date.getTime()) / 1000 / 60 / 60 / 24) * -1;
+    let days = Math.floor((Date.UTC(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate()) - Date.UTC(postedDate.getFullYear(), postedDate.getMonth(), postedDate.getDate()) ) /(1000 * 60 * 60 * 24));
     if(days == 1){
-      return `${days} day ago`
+      return `${days} day ago`;
     }
-    return `${days} days ago`
+    return `${days} days ago`;
+  }
+
+  upvote(index: number): void {
+    this.posts[index].rating += 1;
+    this.loginService.updatePost(this.posts[index]).subscribe(
+      ()=>{},
+      (err) => this.handleError(err.error.Error[0])
+    );
+    while(this.posts[index - 1].rating < this.posts[index].rating){
```

Comment:
What happens if I upvote a post at index 0? I think this might throw an index out of range error since we are doing `this.posts[index-1]`